### PR TITLE
[MM-47560]: In-product: Add support for new fields (back-end and front-end)

### PR DESCRIPTION
**Pull request by emmyni**
#### Summary

Add support for the crossSellsTo field in the Product struct

Related PR: https://github.com/mattermost/mattermost-webapp/pull/11379

#### Ticket Link

[MM-47560](https://mattermost.atlassian.net/browse/MM-47560)
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: model/cloud.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type Product struct {
 	Family            SubscriptionFamily `json:"product_family"`
 	RecurringInterval RecurringInterval  `json:"recurring_interval"`
 	BillingScheme     BillingScheme      `json:"billing_scheme"`
+	CrossSellsTo      string             `json:"cross_sells_to"`
```
Comment (neallred) [inline model/cloud.go:55]: I think we'll also want to add CrossSellsTo to `UserFacingProduct`, a simplified version of the same data that we use in some cases to display product related things to non-admin users. I attached a video of an example of when non-admin users see the pricing modal. I'm imagining we'll also want to let them see the annual vs monthly toggle that admin users would be able to see, so I think we'd also need the `CrossSellsTo` attribute for that.

https://user-images.githubusercontent.com/13738432/196480265-dd89ff07-5aac-496d-ba7e-09fe9fa61134.mp4

**Review by neallred — CHANGES_REQUESTED** [review]
Mostly looks good to me. I think we also want to add this property to `UserFacingProduct`, which ends up getting used in the api handler: https://github.com/mattermost/mattermost-server/blob/master/api4/cloud.go#L301

File: model/cloud.go (hunk 1 above)
Comment (neallred) [reply model/cloud.go:55]: Good question! Yes, I think `RecurringInterval` should also be added. I don't see others that need to be added at this time.

Comment (emmyni) [reply model/cloud.go:55]: Great. thanks :)

**Review by neallred — APPROVED** [review]
LGTM, thanks @emmyni !

File: api4/cloud_test.go [hunk 2]
```diff
@@ -478,27 +480,34 @@ func TestGetCloudProducts(t *testing.T) {
...
 		{
-			ID:           "prod_test1",
-			Name:         "name",
-			PricePerSeat: 10,
-			SKU:          "sku",
+			ID:                "prod_test1",
+			Name:              "name",
+			PricePerSeat:      10,
+			SKU:               "sku",
+			RecurringInterval: "recurring_interval",
```
Comment (nickmisasi) [inline api4/cloud_test.go:493]: Maybe set these to either `monthly` or `yearly` for posterity

**Pull request merged**